### PR TITLE
Fixes #2238 : Ensure resolve requests are listened to

### LINKED ISSUE #2238: excessive warning message that could fill logs
The following log message appears that may fill up logs during high loads:

```
2018-01-03 11:24:08.840  WARN   --- [t_ack_stage, 0)] .e.c.c.i.s.SimpleClusterTierClientEntity: Ignoring the response org.ehcache.clustered.common.internal.messages.EhcacheEntityResponse$ResolveRequest@63207eed as no registered response listener could be found.
```
And from the looks of it, it is also expected behaviour, so the warning message is misleading.

### PR DESCRIPTION
**Pull request by chrisdennis**

### CODE REVIEW

File: clustered/client/src/test/java/org/ehcache/clustered/client/BasicClusteredCacheTest.java [hunk 1]
```diff
@@ -233,6 +233,25 @@ public void testLargeValues() throws Exception {
...
+  public void testClusteredCacheForcingChainResolution() throws Exception {
+    final CacheManagerBuilder<PersistentCacheManager> clusteredCacheManagerBuilder =
+      newCacheManagerBuilder()
+        .with(cluster(CLUSTER_URI).autoCreate())
+        .withCache("clustered-cache", newCacheConfigurationBuilder(Long.class, String.class,
+          ResourcePoolsBuilder.newResourcePoolsBuilder()
+            .with(clusteredDedicated("primary-server-resource", 2, MemoryUnit.MB)))
+          .add(new ClusteredStoreConfiguration(Consistency.STRONG)));
+
+    try (PersistentCacheManager cacheManager = clusteredCacheManagerBuilder.build(true)) {
```
Comment (henri-tremblay) [inline clustered/client/src/test/java/org/ehcache/clustered/client/BasicClusteredCacheTest.java]: I don't get it. How do you test it is forcing chain resolution?

File: clustered/client/src/test/java/org/ehcache/clustered/client/BasicClusteredCacheTest.java [hunk 2]
```diff
@@ -233,6 +233,25 @@ public void testLargeValues() throws Exception {
...
 
+  @Test
+  public void testClusteredCacheForcingChainResolution() throws Exception {
+    final CacheManagerBuilder<PersistentCacheManager> clusteredCacheManagerBuilder =
+      newCacheManagerBuilder()
+        .with(cluster(CLUSTER_URI).autoCreate())
+        .withCache("clustered-cache", newCacheConfigurationBuilder(Long.class, String.class,
+          ResourcePoolsBuilder.newResourcePoolsBuilder()
+            .with(clusteredDedicated("primary-server-resource", 2, MemoryUnit.MB)))
+          .add(new ClusteredStoreConfiguration(Consistency.STRONG)));
```
Comment (henri-tremblay) [inline clustered/client/src/test/java/org/ehcache/clustered/client/BasicClusteredCacheTest.java]: There is a builder for that: `.add(ClusteredStoreConfigurationBuilder.withConsistency(Consistency.STRONG)))`

File: clustered/client/src/test/java/org/ehcache/clustered/client/internal/store/CommonServerStoreProxyTest.java [hunk 3]
```diff
@@ -19,20 +19,25 @@
...
 import java.util.Iterator;
 
 import static org.ehcache.clustered.common.internal.store.Util.createPayload;
 import static org.ehcache.clustered.common.internal.store.Util.getChain;
 import static org.ehcache.clustered.common.internal.store.Util.readPayLoad;
+import static org.ehcache.clustered.client.internal.store.ClusterTierClientEntity.ResponseListener;
 import static org.hamcrest.Matchers.is;
 import static org.junit.Assert.assertThat;
-import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.*;
```
Comment (henri-tremblay) [inline clustered/client/src/test/java/org/ehcache/clustered/client/internal/store/CommonServerStoreProxyTest.java]: Star import (even though I like star imports for static imports.... But it's not the Ehcache style)

File: clustered/client/src/test/java/org/ehcache/clustered/client/internal/store/CommonServerStoreProxyTest.java [hunk 4]
```diff
@@ -19,20 +19,25 @@
 import org.ehcache.clustered.client.config.builders.ClusteredResourcePoolBuilder;
 import org.ehcache.clustered.client.internal.store.ServerStoreProxy.ServerCallback;
 import org.ehcache.clustered.common.internal.ServerStoreConfiguration;
+import org.ehcache.clustered.common.internal.messages.EhcacheEntityResponse;
 import org.ehcache.clustered.common.internal.store.Chain;
 import org.ehcache.clustered.common.internal.store.Element;
 import org.ehcache.config.units.MemoryUnit;
 import org.ehcache.impl.serialization.LongSerializer;
 import org.junit.Test;
+import org.mockito.ArgumentCaptor;
```
Comment (henri-tremblay) [inline clustered/client/src/test/java/org/ehcache/clustered/client/internal/store/CommonServerStoreProxyTest.java]: Where is the captor used?

File: clustered/client/src/test/java/org/ehcache/clustered/client/BasicClusteredCacheTest.java (hunk 1 above)
Comment (chrisdennis) [reply clustered/client/src/test/java/org/ehcache/clustered/client/BasicClusteredCacheTest.java]: This shouldn't still be here.  The test in CommonServerStoreProxyTest covers this.

File: clustered/client/src/test/java/org/ehcache/clustered/client/internal/store/CommonServerStoreProxyTest.java (hunk 4 above)
Comment (chrisdennis) [reply clustered/client/src/test/java/org/ehcache/clustered/client/internal/store/CommonServerStoreProxyTest.java]: Rogue import from earlier experimentation.

**Comment by chrisdennis** [comment]
Test this please

**Pull request merged**